crates/core/src/client/mod.rs

```diff
@@ -631,6 +631,23 @@ impl KakarotClient for KakarotClientImpl<JsonRpcClient<HttpTransport>> {
         Ok(Address::from(evm_address_sliced))
     }
 
+    /// Get the nonce for a given ethereum address
+    /// Convert Ethereum address to Starknet address
+    /// Return the nonce, by calling `get_nonce` for the addresss via starknet rpc
+    /// * `ethereum_address` - The EVM address to get the nonce of
+    /// * `block_id` - The block to get the nonce at
+    ///
+    /// ### Returns
+    /// * `Result<U256, KakarotClientError>` - The nonce of the EVM address
+    async fn nonce(&self, ethereum_address: Address, block_id: StarknetBlockId) -> Result<U256, KakarotClientError> {
+        let starknet_address = self.compute_starknet_address(ethereum_address, &block_id).await?;
+
+        let nonce_felt = self.inner.get_nonce(block_id, starknet_address).await?;
```

Comment:
no need to name with type suffixed. Rust encourages to do 

```
let nonce = self.inner ...
let nonce = U256::from_be_bytes(nonce.to_bytes_be());
```

Reply:
Yeah. I know shadowing is allowed, I guess a type suffix just makes it clear to read, like:
- Oh, yes it was a felt { without going to definition of `compute_starknet_address` }
- This is why we are now converting to a U256

I know code editors can show the type for upper `nonce`, but in absence of them, it makes it more readable.

What say?

Reply:
I'd remove this as types as supposed to be obvious, and the rust book says not to do this naming thing, so let's stay close to it

Reply:
Gotcha!

Reply:
done [here](https://github.com/kkrt-labs/kakarot-rpc/pull/181/commits/cf46bdd8e5adc038658cf7783c2a8bef1f9f8311).